tools/upload.py

```diff
@@ -1,102 +1,64 @@
 #!/usr/bin/env python
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import json
 import os
 from lib.github import GitHub
+from lib.helpers import *
 import requests
 
-BROWSER_LAPTOP_REPO = 'brave/browser-laptop'
 TARGET_ARCH= os.environ['TARGET_ARCH'] if os.environ.has_key('TARGET_ARCH') else 'x64'
 
 def main():
-  print('Running upload...')
-  github = GitHub(auth_token())
-  releases = github.repos(BROWSER_LAPTOP_REPO).releases.get()
-  tag = 'v' + json.load(open('package.json'))['version']
-  tag_exists = False
-  for release in releases:
-    if not release['draft'] and release['tag_name'] == tag:
-      tag_exists = True
-      break
-  release = create_or_get_release_draft(github, releases, tag,
-                                        tag_exists)
-  for f in get_files_to_upload():
-    upload_browser_laptop(github, release, f)
+  print('[INFO] Running upload...')
+  repo = GitHub(get_env('GITHUB_TOKEN')).repos(BROWSER_LAPTOP_REPO)
 
-def get_channel_display_name():
-  d = {'dev': 'Release', 'beta': 'Beta', 'developer': 'Developer', 'nightly': 'Nightly'}
-  return d[release_channel()]
+  sanity_check(repo, get_tag())
 
-def get_files_to_upload():
-  matches = []
-  for root, dirnames, filenames in os.walk('dist'):
-    for filename in filenames:
-      matches.append(os.path.join(root, filename))
-  return matches
+  release = create_release_draft(repo, get_tag())
+  print('[INFO] Uploading release {}'.format(release['tag_name']))
+  for f in get_files_to_upload():
+    upload_browser_laptop(repo, release, f)
+  print('[INFO] Finished upload')
 
+def sanity_check(repo, tag):
+  releases = get_releases_by_tag(repo, tag, include_drafts=True)
+  print('[INFO] Existing releases: {}'.format([x['tag_name'] for x in releases]))
+  if releases:
+    raise(UserWarning("Draft with tag {} already exists".format(tag)))
 
 def upload_browser_laptop(github, release, file_path):
-  filename = os.path.basename(file_path)
-  print('upload_browser_laptop: ' + filename)
-  try:
-    for asset in release['assets']:
-      if asset['name'] == filename:
-        github.repos(BROWSER_LAPTOP_REPO).releases.assets(asset['id']).delete()
```

Comment:
This logic is missing from the new script and is definitely important. There are times where a file gets partially uploaded and then network connection drops. When the build/upload job is re-triggered, it needs to wipe out the files on GitHub

Reply:
Thinking here is if the file exists something went wrong (network or otherwise) and we would want manual intervention.

Maybe makes sense to add retry to the upload?

Reply:
@RyanJarv would definitely be great to have a retry built into the upload script- right now the retry is done manually... Opening up Jenkins and re-queueing the failed build type (ex: win32). When that is manually re-queued, the delete is needed (otherwise GitHub will return an error since the file exists- which may only be a partial upload)

Reply:
Here are two examples from this morning where the upload failed:
https://jenkins.brave.com/job/browser-laptop-build-win32/426/console
https://jenkins.brave.com/job/browser-laptop-build-win64/423/console

Reply:
@bsclifton added some retry logic if you have a sec to take a look at this again